gson/src/main/java/com/google/gson/Gson.java

```diff
@@ -512,6 +518,36 @@ static void checkValidFloatingPoint(double value) {
   }
 
   /**
+   * This method is used to get the {@link TypeAdapter} for the specified field. The type
+   * adapter needs to be configured by {@link GsonBuilder#registerFieldAdapterFactory}.
+   *
+   * @param field the field needs to be investigated
+   * @return the {@link TypeAdapter} for the given {@code field}
+   * @since 2.8.1
+   */
+  public <T> TypeAdapter<T> getFieldAdapter(Field field) {
+    if (fieldAdapterFactories == null || fieldAdapterFactories.isEmpty()) {
+      return null;
+    }
+    for (Annotation fieldAnnotation : field.getAnnotations()) {
+      Class<? extends Annotation> annotationType = fieldAnnotation.annotationType();
+      if (fieldAdapterFactories.containsKey(annotationType)) {
+        FieldAdapterFactory fieldAdapterFactory = fieldAdapterFactories.get(annotationType);
+        return fieldAdapterFactory.create(fieldAnnotation, field);
+      }
+    }
+    // If both this field and its declaring class matched, ignore the latter one
+    for (Annotation classAnnotation : field.getDeclaringClass().getAnnotations()) {
```

Comment:
Removed the 'misleading' comment, but I think the logic here is natural - firtst look at the field annotation, and then the class annotations...